django/db/models/sql/compiler.py

```diff
@@ -357,8 +357,8 @@ def get_order_by(self):
 
     def get_extra_select(self, order_by, select):
         extra_select = []
-        select_sql = [t[1] for t in select]
         if self.query.distinct and not self.query.distinct_fields:
+            select_sql = [t[1] for t in select]
```

Comment:
looks to me like it could actually be a set comprehension for more speed again since it's only used for N `not in` checks below

Reply:
`select` is a list of tuples `(sql, params)` where `params` is a list, therefore I think we cannot use `set` in this case.

Reply:
fair enough, I was just taking a very narrow view here